src/main/java/org/team1540/robot2025/subsystems/elevator/Elevator.java

```diff
@@ -0,0 +1,140 @@
+package org.team1540.robot2025.subsystems.elevator;
+
+import static org.team1540.robot2025.subsystems.elevator.ElevatorConstants.*;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.RobotState;
+import edu.wpi.first.wpilibj.XboxController;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.Subsystem;
+import org.littletonrobotics.junction.AutoLogOutput;
+import org.littletonrobotics.junction.Logger;
+import org.team1540.robot2025.Constants;
+import org.team1540.robot2025.util.MechanismVisualiser;
+import org.team1540.robot2025.util.math.AverageFilter;
+
+import java.util.function.DoubleSupplier;
+
+import java.util.DoubleSummaryStatistics;
+import java.util.function.DoubleSupplier;
+
+public class Elevator implements Subsystem {
+    private final ElevatorIO io;
+    private final ElevatorIOInputsAutoLogged inputs = new ElevatorIOInputsAutoLogged();
+    private final AverageFilter positionFilter = new AverageFilter(10);
+    private double setpointMeters;
+    private final double DEADZONE = 0.2;
+
+    private static boolean hasInstance = false;
+
+    private Elevator(ElevatorIO elevatorIO) {
+        if (hasInstance) throw new IllegalStateException("Instance of elevator already exists");
+        hasInstance = true;
+        this.io = elevatorIO;
+    }
+
+    public static Elevator createReal() {
+        if (Constants.currentMode != Constants.Mode.REAL) {
+            DriverStation.reportWarning("Using real elevator on simulated robot", false);
+        }
+        return new Elevator(new ElevatorIOTalonFX());
+    }
+
+    public static Elevator createSim() {
+        if (Constants.currentMode == Constants.Mode.REAL) {
+            DriverStation.reportWarning("Using simulated elevator on real robot", false);
+        }
+        return new Elevator(new ElevatorIOSim());
+    }
+
+    public static Elevator createDummy() {
+        if (Constants.currentMode == Constants.Mode.REAL) {
+            DriverStation.reportWarning("Using dummy elevator on real robot", false);
+        }
+        return new Elevator(new ElevatorIO() {});
+    }
```

Comment:
The factory boilerplate probably wants to go at the end of the file so we don't have to scroll past it every time we want to read this file. (i put this at the beginning of the file last year and it was annoying, would not recommend)